### DISCUSSION TITLE
How to rebuild provider without watch?

### DISCUSSION THREAD

**Discussion by vimhead** [rrousselGit/riverpod#971] [Q&A] [answered]
I want to rebuild provider without ref.watch(anotherProvider). Is that possible? 
```
Provider((ref) {
 ref.watch(anotherProvider);
 return Repository();
});
```
seems a bit hacky.

**Comment by rrousselGit** [comment, marked as answer]
Could you explain why?

`ref.watch` is the expected solution.

**Reply by vimhead** [reply]
I thought that ref.watch return value is supposed to be used. Thank you.

**Reply by rrousselGit** [reply]
That's not necessary, no.

Calling `ref.watch` without using the return value is a useful pattern

**Reply by vimhead** [reply]
Thank you for awesome package. Have a good day!
